dragonfly/player/player.go

```diff
@@ -1503,6 +1539,29 @@ func (p *Player) Tick(current int64) {
 		p.Hurt(4, damage.SourceVoid{})
 	}
 
+	if p.OnFireDuration() > 0 {
+		if p.FireProof() {
+			p.SetOnFire(0)
+		} else {
+			p.SetOnFire(p.OnFireDuration() - time.Second/20)
```

Comment:
Could only subtract one from the player's current fire ticks and only call Extinguish() if they drop below 0 so you don't send packets everytime.